File: SDKs/twilio.com/twilio_sync_v1/1.36.0/python/src/sdk/models/operations/updatedocumentpermission.py
from dataclasses import dataclass, field
from typing import Optional
from sdk.models import shared
UPDATE_DOCUMENT_PERMISSION_SERVERS = [
	"https://sync.twilio.com",
]


@dataclass
class UpdateDocumentPermissionPathParams:
    document_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'DocumentSid', 'style': 'simple', 'explode': False }})
    identity: str = field(default=None, metadata={'path_param': { 'field_name': 'Identity', 'style': 'simple', 'explode': False }})
    service_sid: str = field(default=None, metadata={'path_param': { 'field_name': 'ServiceSid', 'style': 'simple', 'explode': False }})
    

@dataclass
class UpdateDocumentPermissionRequestBodyUpdateDocumentPermissionRequest:
    manage: bool = field(default=None, metadata={'form': { 'field_name': 'Manage' }})
    read: bool = field(default=None, metadata={'form': { 'field_name': 'Read' }})
    write: bool = field(default=None, metadata={'form': { 'field_name': 'Write' }})
    

@dataclass
class UpdateDocumentPermissionSecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class UpdateDocumentPermissionRequest:
    server_url: Optional[str] = field(default=None)
    path_params: UpdateDocumentPermissionPathParams = field(default=None)
    request: Optional[UpdateDocumentPermissionRequestBodyUpdateDocumentPermissionRequest] = field(default=None, metadata={'request': { 'media_type': 'application/x-www-form-urlencoded' }})
    security: UpdateDocumentPermissionSecurity = field(default=None)
    

@dataclass
class UpdateDocumentPermissionResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    sync_v1_service_document_document_permission: Optional[shared.SyncV1ServiceDocumentDocumentPermission] = field(default=None)
    

File: SDKs/twilio.com/twilio_sync_v1/1.36.0/typescript/src/sdk/models/operations/updatedocumentpermission.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const UPDATEDOCUMENTPERMISSION_SERVERS = [
	"https://sync.twilio.com",
];



export class UpdateDocumentPermissionPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=DocumentSid" })
  documentSid: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=Identity" })
  identity: string;

  @Metadata({ data: "pathParam, style=simple;explode=false;name=ServiceSid" })
  serviceSid: string;
}


export class UpdateDocumentPermissionUpdateDocumentPermissionRequest extends SpeakeasyBase {
  @Metadata({ data: "form, name=Manage;" })
  manage: boolean;

  @Metadata({ data: "form, name=Read;" })
  read: boolean;

  @Metadata({ data: "form, name=Write;" })
  write: boolean;
}


export class UpdateDocumentPermissionSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class UpdateDocumentPermissionRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  pathParams: UpdateDocumentPermissionPathParams;

  @Metadata({ data: "request, media_type=application/x-www-form-urlencoded" })
  request?: UpdateDocumentPermissionUpdateDocumentPermissionRequest;

  @Metadata()
  security: UpdateDocumentPermissionSecurity;
}


export class UpdateDocumentPermissionResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  syncV1ServiceDocumentDocumentPermission?: shared.SyncV1ServiceDocumentDocumentPermission;
}
